suite-native/module-accounts-import/src/components/AccountImportDetail.tsx

```diff
@@ -0,0 +1,126 @@
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { Box, Button, Divider, Text } from '@suite-native/atoms';
+import { Form, useForm } from '@suite-native/forms';
+import { AccountsRootState, selectAccountsByNetworkAndDevice } from '@suite-common/wallet-core';
+import { yup } from '@trezor/validation';
+import { networks, NetworkSymbol } from '@suite-common/wallet-config';
+import { AccountInfo } from '@trezor/connect';
+
+import { HIDDEN_DEVICE_ID, HIDDEN_DEVICE_STATE, importAccountThunk } from '../accountsImportThunks';
+import { AccountImportOverview } from './AccountImportOverview';
+
+type AccountImportDetailProps = {
+    networkSymbol: NetworkSymbol;
+    accountInfo: AccountInfo;
+};
+
+const importAnotherButtonStyle = prepareNativeStyle(utils => ({
+    marginTop: utils.spacings.large,
+    marginBottom: utils.spacings.medium,
+}));
+const contentWrapperStyle = prepareNativeStyle(_ => ({
+    width: '100%',
+    flex: 1,
+    justifyContent: 'flex-end',
+}));
+
+const accountImportFormValidationSchema = yup.object({
+    accountLabel: yup.string().required().max(30),
+});
+type AccountImportFormValues = yup.InferType<typeof accountImportFormValidationSchema>;
+
+type NavigationProp = StackToTabCompositeScreenProps<
+    AccountsImportStackParamList,
+    AccountsImportStackRoutes.AccountImport,
+    RootStackParamList
+>;
+
+export const AccountImportDetail = ({ networkSymbol, accountInfo }: AccountImportDetailProps) => {
+    const { applyStyle } = useNativeStyles();
+    const dispatch = useDispatch();
+    const navigation = useNavigation<NavigationProp>();
+    const deviceNetworkAccounts = useSelector((state: AccountsRootState) =>
+        selectAccountsByNetworkAndDevice(state, HIDDEN_DEVICE_STATE, networkSymbol),
+    );
+    const defaultAccountLabel = `${networks[networkSymbol].name} #${
+        deviceNetworkAccounts.length + 1
+    }`;
+    const form = useForm<AccountImportFormValues>({
+        validation: accountImportFormValidationSchema,
+        defaultValues: {
+            accountLabel: defaultAccountLabel,
+        },
+    });
+    const {
+        handleSubmit,
+        formState: { errors },
+    } = form;
+
+    const handleImportAccount = ({ accountLabel }: AccountImportFormValues) => {
+        if (accountInfo) {
+            dispatch(
+                importAccountThunk({
+                    deviceId: HIDDEN_DEVICE_ID,
+                    deviceTitle: 'Hidden Device',
+                    accountInfo,
+                    accountLabel,
+                    coin: networkSymbol,
+                }),
+            );
+            dispatch(setOnboardingFinished(true));
+            navigation.navigate(RootStackRoutes.AppTabs, {
+                screen: AppTabsRoutes.HomeStack,
+                params: {
+                    screen: HomeStackRoutes.Home,
+                },
+            });
+        }
+    };
+
+    const handleImportAccountSubmit = handleSubmit(handleImportAccount);
```

Comment:
nit: Seems little bit redundant since it's only in one place and it's very short, so I would suggest to put it directly into `onPress={handleSubmit(handleImportAccount)}`

Reply:
I think it's better to wrap the callback in handler like this, what do you think? 
[0f0f9b2](https://github.com/trezor/trezor-suite/pull/6591/commits/0f0f9b2143cff7e3085e644f8f26a314314aa6f9)

Reply:
I don't mind, either solution is OK. My is probably little bit more flexible because you can use handler without form then, but it's doesn't happen so much often.